File: TSVersion/Protocol/Response.ts
import {ProxyRequestPattern} from "../ProxySession"


export interface SHResponse{
    type: string
    operation: string
}


export class SHCryptoHandshakeResponse implements SHResponse{
    readonly type: string = "response"
    readonly operation: string = ProxyRequestPattern.handshake
    UUID: string
    key: string
}


export class SHCryptoDataResponse implements SHResponse{
    readonly type: string = "response"
    operation: string
    response: string
}


export class SHDataResponse implements SHResponse{
    readonly type: string = "response"
    operation: string
    result: any
}


export class SHStatusResponse implements SHResponse{
    readonly type: string = "response"
    operation: string
    result: boolean = true
    result_message: string
    description: string
}

File: untitled1/Protocol/response.cpp
#include "response.h"



//struct SHResponse

void SHResponse::read(const QJsonObject &json){
    type = json["type"].toString();
    operation = json["operation"].toString();
}


void SHResponse::write(const QJsonObject &json){
    json["type"] = type;
    json["operation"] = operation;
}



//struct SHCryptoHandshakeResponse: public SHResponse

void SHCryptoHandshakeResponse::read(const QJsonObject &json){
    SHResponse::read(json);
    key = json["key"].toString();
}


void SHCryptoHandshakeResponse::write(const QJsonObject &json){
    SHResponse::write(json);
    json["key"] = key;
}



//struct SHCryptoDataResponse: public SHResponse

void SHCryptoDataResponse::read(const QJsonObject &json){
    SHResponse::read(json);
    response = json["response"].toString();
}


void SHCryptoDataResponse::write(const QJsonObject &json){
    SHResponse::write(json);
    json["response"] = response;
}



//struct SHStatusResponse: public SHResponse

void SHStatusResponse::read(const QJsonObject &json){
    SHResponse::read(json);
    result = json["result"].toBool();
    result_message = json["result_message"].toString();
    description = json["description"].toString();
}


void SHStatusResponse::write(const QJsonObject &json){
    SHResponse::write(json);
    json["result"] = result;
    json["result_message"] = result_message;
    json["description"] = description;
}
